docs/design/multi_slo_doc_support.md

```diff
@@ -0,0 +1,102 @@
+* [Option 1] (least effort): Update the 'ServiceSLO' table to include a 'SLODocName' column of type 'string'.
+* [Option 2] (author's recommendation): Update the 'ServiceSLO' table to include a 'SLODocName' column of type 'integer', which is a foreign key to a new table 'SLODocName'. Table 'SLODocName' contains 'id' (primary key, integer), and 'name' (string, unique)
+    * Better data normalization that Option 1
+    * Less disruptive to schema than Option 3
+* [Option 3] (most effort): Same as Option 2, and in addition: update all tables that currently have a relationship with table 'ServiceSLO' to instead have a relationship with 'SLODocName'
+
+[This PR](https://github.com/app-sre/dashdotdb/pull/50) uses the "Option 2" approach.
+
+This digram visualizes the proposed schema change, introducing the 'SLODocName' table. (bottom left):
+![](images/multi-slo-doc-schema-change.png)
+
+### Other Changes To DashDotDB 
+
+* dash.db
+  * this needs a "slo_doc_name" property: https://github.com/app-sre/dashdotdb/blob/f88048cd9156afa93a4a0448ecd088c551c8408d/dashdotdb/models/dashdotdb.py
+  * update this to include "slo_doc_name": https://github.com/app-sre/dashdotdb/blob/f88048cd9156afa93a4a0448ecd088c551c8408d/examples/serviceslometrics.json
+  * update this image: https://github.com/app-sre/dashdotdb/blob/f88048cd9156afa93a4a0448ecd088c551c8408d/docs/dashdotdb.png (and the diagram file for it: https://github.com/app-sre/dashdotdb/blob/f88048cd9156afa93a4a0448ecd088c551c8408d/docs/dashdotdb.dia)
+  * Follow the [DB upgrade guide](https://github.com/app-sre/dashdotdb#db-upgrade) to upgrade the schema in Postgres
+
+### Changes To Other Systems
+
+The following are changes to related systems that are expected to need to be made.
+
+* app-interface
+  * add "slo_doc_name": https://gitlab.cee.redhat.com/service/app-interface/-/blob/15d1ce504a9233a04d6119e8d57e58a3bcc31cb0/graphql-schemas/schema.yml#L1916-1922
+  * update this to include 'slo_doc_name': https://gitlab.cee.redhat.com/service/app-interface/-/blob/15d1ce504a9233a04d6119e8d57e58a3bcc31cb0/schemas/app-sre/slo-document-1.yml
+  * All of the SLO documents here need to have the new 'slo_doc_name' proeprty set: https://gitlab.cee.redhat.com/service/app-interface/-/tree/5381889ea1aecd1d5afd51fae246c6a5909438ef/data/services
+  * all of these same sort of changes in public-github app-interface
+
+* visual-qontract
+  * This needs to accept MULTIPLE 'slo_documents' as a parameter https://github.com/app-sre/visual-qontract/blob/bb5e993a428780ccd6a1d4208b36740e46d96f79/src/pages/elements/Report.js#L547
+  * This must pass MULTIPLE 'slo_documents': https://github.com/app-sre/visual-qontract/blob/bb5e993a428780ccd6a1d4208b36740e46d96f79/src/pages/elements/Report.js#L704
+  * This needs to be reethought to grab multiple slo documents, rather than just 1: https://github.com/app-sre/visual-qontract/blob/bb5e993a428780ccd6a1d4208b36740e46d96f79/src/pages/elements/Report.js#L633-L643
```

Comment:
Do you have ideas on how to do this?  Being this vague isn't going to helpful to understand what was changed because of this proposal

Reply:
I do need to spend more time on thinking through the code changes for visual-qontract.

My intended scope with this design-doc was primarily just around dashdotDB changes, with some high-level references to related systems.

If desired, I'm happy to take more time to update this document to more holistically cover the specific code changes to all related systems.

Reply:
If it's not related to DDDB, I wouldn't include it tbh.  This design doc should just be about changes to DDDB and independent of other services.  Those other services will need their down design docs for how to consume the changes in DDDB.

Of course, the changes here need to be done in a way that can be consumed by the users of the system, but how they will do it really isn't a concern of the DDDB proposal.

Reply:
Fair enough. I will delete the section with references to high-level changes to other systems from this particular design doc.